frontend/src/components/Main.js

```diff
@@ -1,19 +1,21 @@
-import { Fragment } from 'react'
+import { Fragment, useState } from 'react'
 import SearchForm from './SearchForm';
 import ResultsList from './ResultsList';
 import '../sass/components/main.scss';
 
 function Main() {
-    
-    
-
+    const [resultsData, setResultsData] = useState([]);
 
+    const setResultsDataHandler = (value) => {
```

Comment:
After you apply the suggestions below of sending an object like this `{name: "something"}` and successfully fetch the data, I want you to open your console and see if you will notice anything strange. After that, `import {useCallback} from 'react'` and do this
```javascript
const setResultsDataHandler = useCallback((value) => {
        setResultsData(value);
    }, []);
```
Then open your browser console and fetch the data again and observe if there is any difference.